mkosi/config.py

```diff
@@ -3846,6 +3884,26 @@ def parse_config_one(self, path: Path, parse_profiles: bool = False, parse_local
                             ),
                         )
 
+                for f in s.recursive_paths:
+                    recursive_extras = parse_paths_and_dirs(
+                        f,
+                        secret=s.path_secret,
+                        required=False,
+                        resolve=False,
+                        expanduser=False,
+                        expandvars=False,
+                    )
+                    for e in recursive_extras:
+                        if e.exists():
+                            setattr(
+                                self.config,
+                                s.dest,
+                                s.parse(
+                                    file_run_or_read(e).rstrip("\n") if s.path_read_text else os.fspath(e),
```

Comment:
We should never use `path_read_text` when more than one path can be specified so using `file_run_or_read()` on paths from a directory should never happen in the first place